osx/Updater/AppDelegate.m

```diff
@@ -0,0 +1,56 @@
+//
+//  AppDelegate.m
+//  Updater
+//
+//  Created by Gabriel on 4/7/16.
+//  Copyright © 2016 Gabriel Handford. All rights reserved.
+//
+
+#import "AppDelegate.h"
+
+#import "Defines.h"
+#import "Prompt.h"
+
+@implementation AppDelegate
+
+- (void)applicationDidFinishLaunching:(NSNotification *)notification {
+  dispatch_async(dispatch_get_main_queue(), ^{
+    [self run];
+  });
+}
+
+- (void)exitWithError:(NSError *)error {
+  NSLog(@"Error: %@", error.localizedDescription);
+  fflush(stderr);
```

Comment:
Yep refactored to have a single exit method. Out and err are flushed on all exits.